omigami/spectra_matching/spectra_matching.py

```diff
@@ -183,41 +188,71 @@ def _make_batch_requests(
         parameters: SpectraMatchingParameters,
         endpoint: str,
     ) -> List[pd.DataFrame]:
+        self._failed_spectra = []
         batch = []
-        requests = []
+        results = []
 
         for spectrum in spectra_generator:
+            cache_id = self._get_cache_id(spectrum)
+            if cache_id in self._cached_results:
+                results.append(self._cached_results[cache_id])
+                continue
+
             batch.append(spectrum)
             if len(batch) == SPECTRA_LIMIT_PER_REQUEST:
-                payload = self._build_payload(batch, parameters)
-                requests.append(self._send_request(payload, endpoint))
+                results.extend(self._match_spectra(batch, endpoint, parameters))
                 batch = []
 
-        if batch:
-            payload = self._build_payload(batch, parameters)
-            requests.append(self._send_request(payload, endpoint))
+        if len(batch) > 0:
+            results.extend(self._match_spectra(batch, endpoint, parameters))
 
-        predictions = []
-        for r in requests:
-            predictions.extend(self._format_results(r))
-        return predictions
+        return results
+
+    def _match_spectra(
+        self,
+        batch: List[Spectrum],
+        endpoint: str,
+        parameters: SpectraMatchingParameters,
+    ) -> List[pd.DataFrame]:
+        response = self._send_request(batch, endpoint, parameters)
+        if response is not None:
+            formatted_results = self._format_results(response)
+            self._cache_results(formatted_results, batch)
```

Comment:
Doesn't seem like you are using the SpectraMatchingParameters in hashing, but we should, right?

Reply:
true I forgot that we talked about this